CONTRIBUTING.md

```diff
@@ -0,0 +1,331 @@
+message when merging the request.
+
+The title should therefore be less than about 70 characters long and
+describe the change made by the pull request in a terse way.
+
+The description should describe in detail what the pull request changes.
+If the pull request results in a breaking change, the description should
+note that.
+
+Do not update `Changelog.md` as part of the pull request. Git tends to
+mess up merging this file so there will weird merge conflicts for every
+request. Instead, the person merging will update the changelog.
+
+## Code organization and formatting
+
+The following describes the principles on which the current structure of
+the code is based. These aren’t hard and fast rules that need to be
+followed at all cost but rather describe how things would be handled in an
+ideal world. If there is good reasons to do things differently, then do
+so.
+
+The overall purpose is to make it as easy as possible for a reader to find
+what they are looking for and understand how things work.
+
+As a consequence, only few of the following is enforced formally (the
+`rustfmt` run as part of the CI workflow being the possibly only
+exceptions). But in a review of a pull request, there may be questions …
+
+### Repository structure
+
+#### Branches and tags
+
+The _main_ branch contains the current status of the crate. It has to be
+kept in a state that builds cleanly as indicated by the CI workflow
+succeeding. If the branch is ‘broken,’ it should be fixed as soon as
+possible. As a consequence, it should generally be okay to use this branch
+as a git dependency in a Cargo.toml.
+
+Each release version is tagged with an annotated tag named `v0.x.y`. This tag
+is referencing the merge commit of the PR for that release.
+
+If changes need to be backported to an older version of the crate, start a
+`series-0.x` branch from the release tag of the last release of that series.
+This branch can be deleted once a release with the backported changes has
+been made.
+
+All other branches that do not have a pull request associated with them
+can and should be deleted to keep the repository clean. Use a draft PR to
+signal that you want to keep a branch around.
+
+#### Changelog
+
+The repository contains a dedicated `Changelog.md` that is updated
+manually by whoever merges a pull request and forms the basis for the
+release notes. We keep it as a single file with all the changes in the
+repository to make is easier to hunt for the time of a certain change
+without having to consult Github – or indeed the repository being hosted
+on Github for all eternity. It is written manually and not auto-generated
+from commit message because what should go into the two differs. The
+changelog entry contains a short summary of what changed, the commit
+message should be more elaborate and possibly also explain the reasons.
+
+### Code structure
+
+#### Module structure
+
+Code within a module is grouped by “primary,” i.e., function or type.
```

Comment:
This may be Rust specific (knowledge I lack), but 'grouped by "primary"' is confusing to me. Isn't a (multi-)level grouping always about "primary" and "secondary"?

Reply:
"grouped in blocks, hereafter called .."?

Reply:
The intention is to say that everything for a given type is kept together rather than first doing all the type definitions and then all the implementation blocks.